### ISSUE TITLE
After update I have RangeError: Maximum call stack size exceeded.

### ISSUE THREAD

**Issue by xuxicheta** [JsDaddy/ngx-mask#597]
Here is a debug message
```
ERROR Error: Uncaught (in promise): RangeError: Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:191)
    at SafeSubscriber.next (Subscriber.js:122)
    at Subscriber._next (Subscriber.js:72)
    at Subscriber.next (Subscriber.js:49)
    at EventEmitter.next (Subject.js:39)
    at EventEmitter.emit (core.js:35297)
    at FormControl.updateValueAndValidity (forms.js:4057)
    at FormControl.setValue (forms.js:4692)
    at updateControl (forms.js:3299)
    at MaskService.onChange (forms.js:3271)
    at resolvePromise (zone-evergreen.js:793)
    at new ZoneAwarePromise (zone-evergreen.js:875)
    at __awaiter (tslib.es6.js:69)
    at MaskDirective.writeValue (ngx-mask.js:1006)
    at forms.js:3316
    at forms.js:4690
    at Array.forEach (<anonymous>)
    at FormControl.setValue (forms.js:4686)
    at ngx-formly-core.js:906
    at SafeSubscriber.schedulerFn [as _next] (core.js:35378)
``` 

It call `writeValue` many times with the same `inputValue`

**Comment by intellix** [comment]
Getting this now when coupled with Formly 5.5.0, which I can see you're using also. When we downgrade, it's fine.

**Comment by aitboudad** [comment]
I've done a fix which would be available in `v5.5.1`, but I think `ngx-mask` should fix the following case: when passing `emitEvent` as `false`, it should not emit a new value except when changed:

```
control.setValue(value, {emitEvent: false});
```

**Comment by aitboudad** [comment]
the fix is now part of [v5.5.1](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.5.1) release!

**Comment by isysenko** [comment]
@xuxicheta HI, do you still have error with v5.5.1 version of ngx-formly?

**Comment by xuxicheta** [comment]
@isysenko loooks like fixed. Everything is ok.

**Comment by intellix** [comment]
we're still getting this heavily actually

**Comment by aitboudad** [comment]
@intellix could you please fill an issue in `formly` with a reproduction example, thanks!

**Comment by aitboudad** [comment]
@intellix the only case that may happen in our case is when multiple fields use the same key which I've pushed another fix in [v5.5.7](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.5.7) release.
